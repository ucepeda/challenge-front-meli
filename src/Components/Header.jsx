import React, { useEffect } from "react";
import { useNavigate, useLocation } from "react-router-dom";
import logo from './../assets/Logo_ML.png';
import searchIcon from './../assets/ic_Search.png';
import * as C from "./../utils/constants";
import { Stack, Button, TextField } from '@mui/material/'
import "./../css/globals.scss"


/* Search function */
const Header = ({ handleSearch }) => {
  const navigate = useNavigate();
  const location = useLocation();
  const search = (e) => {
    const [{ value }] = e.target;
    e.preventDefault();
    // prevent empty search
    if (value.trim() !== "") {
      handleSearch(value);
      navigate(`${C.SEARCH_URL}${value}`);
    }
  };

  useEffect(() => {
    const [_, product] = location.search.split("=");
    product ? handleSearch(product) : null;
  }, []);

  return (
    <>
      <form onSubmit={search}>
        <Stack
          className="header-main"
          direction="row"
          justifyContent="center"
          alignItems="center"
        >
          <img
            className="logo-meli"
            alt={C.LOGO_ALT_ICO}
            src={logo}
            onClick={() => navigate("/")}
          />
          <TextField
            className="input-search"
            id="search"
            name="search"
            size="small"

            placeholder={C.SEARCH_PLACEHOLDER}
          />
          <Button type="submit" className="btn-search">
            <img
              src={searchIcon}
              alt={C.SEARCH_ALT_ICO}
            />
          </Button>
        </Stack>
      </form>
    </>
  );
};

export default Header;