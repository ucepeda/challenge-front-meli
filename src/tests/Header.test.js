import { BrowserRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils'
import { render } from "@testing-library/react";
import Header from "./../Components/Header";

describe("Header", () => {
    test("should render Header", async () => {
        const component = await act(async () => render(
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        ));
        expect(component).toMatchSnapshot();
    });
});