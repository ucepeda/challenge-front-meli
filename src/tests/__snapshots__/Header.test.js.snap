// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="header-main css-7owdng-MuiStack-root"
        >
          <img
            alt="Logo"
            class="logo-meli"
            src="test-file-stub"
          />
          <div
            class="MuiFormControl-root MuiTextField-root input-search css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="search"
                name="search"
                placeholder="Nunca dejes de buscar"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn-search css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            <img
              alt="Search icon"
              src="test-file-stub"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="header-main css-7owdng-MuiStack-root"
      >
        <img
          alt="Logo"
          class="logo-meli"
          src="test-file-stub"
        />
        <div
          class="MuiFormControl-root MuiTextField-root input-search css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="search"
              name="search"
              placeholder="Nunca dejes de buscar"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium btn-search css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <img
            alt="Search icon"
            src="test-file-stub"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
